SmartDeviceLink_Example/ProxyManager.swift

```diff
@@ -0,0 +1,278 @@
+    func startManager() {
+        sdlManager.start(readyHandler: { [unowned self] (success, error) in
+            guard success else {
+                SDLLog.e("There was an error while starting up: \(String(describing: error))")
+                self.resetConnection()
+                return
+            }
+
+            self.delegate?.didChangeProxyState(SDLProxyState.connected)
+
+            self.buttonManager = ButtonManager(sdlManager: self.sdlManager, updateScreenHandler: self.refreshUIHandler)
+            self.vehicleDataManager = VehicleDataManager(sdlManager: self.sdlManager, refreshUIHandler: self.refreshUIHandler)
+
+            RPCPermissionsManager.setupPermissionsCallbacks(with: self.sdlManager)
+
+            SDLLog.d("SDL file manager storage: \(self.sdlManager.fileManager.bytesAvailable / 1024 / 1024) mb")
+        })
+    }
+}
+
+// MARK: - SDLManagerDelegate
+
+extension ProxyManager: SDLManagerDelegate {
+    /// Called when the connection beween this app and SDL Core has closed.
+    func managerDidDisconnect() {
+        delegate?.didChangeProxyState(SDLProxyState.stopped)
+        firstHMILevelState = .none
+
+        // If desired, automatically start searching for a new connection to Core
+        if ExampleAppShouldRestartSDLManagerOnDisconnect.boolValue {
+            startManager()
+        }
+    }
+
+    /// Called when the state of the SDL app has changed. The state limits the type of RPC that can be sent. Refer to the class documentation for each RPC to determine what state(s) the RPC can be sent.
+    ///
+    /// - Parameters:
+    ///   - oldLevel: The old SDL HMI Level
+    ///   - newLevel: The new SDL HMI Level
+    func hmiLevel(_ oldLevel: SDLHMILevel, didChangeToLevel newLevel: SDLHMILevel) {
+        if newLevel != .none && firstHMILevelState == .none {
+            // This is our first time in a non-NONE state
+            firstHMILevelState = newLevel
+
+            // Send static menu items. Menu related RPCs can be sent at all `hmiLevel`s except `NONE`
+            createStaticMenus()
+            vehicleDataManager.subscribeToVehicleOdometer()
+        }
+
+        if newLevel == .full && firstHMILevelState != .full {
+            // This is our first time in a `FULL` state.
+            firstHMILevelState = newLevel
+        }
+
+        switch newLevel {
+        case .full:                // The SDL app is in the foreground
+            // Always try to show the initial state to guard against some possible weird states. Duplicates will be ignored by Core.
+            showInitialData()
+        case .limited: break        // The SDL app's menu is open
```

Comment:
Limited is when an active NAV / MEDIA app is in the background (and may still be controlled even though its UI isn't onscreen).

Reply:
Fix in both Swift and Obj-C Ex. apps